queries.md

```diff
@@ -4,40 +4,146 @@
 
 ### 1. All the companies that it's name match 'Babelgum'. Retrieve only their `name` field.
 
+filter: { name: {$eq: "Babelgum"}}
+project: {name: 1}
+
 ### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
 
-### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fileds.
+filter: { number_of_employees: {$gte: 5000}}
+sort: {number_of_employees : -1}
+limit: 20
 
+### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fileds.
+filter: { founded_year: {$gte: 2000, $lte: 2005}}  
+project: {name: 1, founded_year: 1, _id: 0}
 ### 4. All the companies that had an IPO of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
+filter: {$and: [{'ipo.valuation_amount': {$gt: 100000000}}, {founded_year: {$lte: 2010}}]}
+project: {name: 1}, {ipo: 1}
 
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
 
+filter: {number_of_employees: {$lt: 1000}, founded_year: {$lt: 2005}}
+sort: {number_of_employees: 1} 
+limit: 10
+
 ### 6. All the companies that don't include the `partners` field.
 
+filter: { partners: {$exists: false}}
+sort: 
+limit:
+
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
+filter: {category_code: {$type: "null"}}
+sort: 
+limit:
+
+
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
+filter: {number_of_employees: {$gte: 100}, number_of_employees: {$lt: 1000}}
+project: {name: 1}, {number_of_employees: 1}
+sort: 
+skip: 
+limit: 
+
+
 ### 9. Order all the companies by their IPO price descendently.
 
+filter: 
+project:
+sort: {'ipo.valuation_amount': -1}
+skip:
+limit:
+
+
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
+filter: 
+project:
+sort: {number_of_employees: -1}
+skip:
+limit: 10
+
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
+filter: {founded_month: {$gte: 6}}
+project:
+sort: 
```

Comment:
gte takes "6" into account, but we want companies founded from July and further, so it would be either:
`{ $gt: 6 }`
or
`{ $gte: 7 }`